proposals/lip-0038.md

```diff
@@ -102,6 +102,8 @@ Here, ceiling is an auxiliary function defined below.
 def verifyAggSig(keysList: list[bytes], aggregationBits: bytes, signature: bytes, tag: bytes, chainID: bytes, message: bytes) -> bool:
     taggedMessage = tagMessage(tag, chainID, message)
     keys = []
+    if len(aggregationBits) != ceiling(len(keysList), 8):
+        return False
```

Comment:
shouldn't we check if there are more bit on aggregationBits, they must be all zero?

Reply:
@shuse2 I was following Jan's idea "Ensure that aggregation bits is exactly ceil( # of BLS keys / 8) and for all bits without corresponding key the bits must be 0." I suppose you're relating to the second part of his sentence but I wasn't sure why it is necessary: if we check that `len(aggregationBits) == ceiling(len(keysList), 8)`,  how it can be more bits? I'm missing some part of understading here, please explain... thanks

Reply:
@shuse2 is right. It could be, for example, that `len(keysList) = 5`. In this case, the "last" 3 bits of the single byte in `aggregationBits` must be zero.

Reply:
Ah, OK. Thanks. I'm not used to bytes and bits, so it took me some time to understand. I'll push the solution I think works.

Reply:
@shuse2 @AndreasKendziorra Before I change the code: since there are four occurences of the code I'm about to enter (twice in LIP 0038 and twice in LIP 0062), can you please check if this is correct beforehand?
```
extra_bits = len(aggregationBits) * 8 - len(keysList)
    if extra_bits > 0:
        last_byte = aggregationBits[-1]
    for i in range(extra_bits):
        bit_position = 8 - extra_bits + i
        bit = (last_byte // (2 ** bit_position)) % 2
        if bit != 0:
            return False
```

Reply:
Wow, thanks. Looks easy :)